File: java-vsc/problems/1221.分割平衡字符串.java
/*
 * @lc app=leetcode.cn id=1221 lang=java
 *
 * [1221] 分割平衡字符串
 */

// @lc code=start
class Solution {
    public int balancedStringSplit(String s) {
        int balanceCount = 0;
        int len = s.length();
        int lCnt = 0;
        int rCnt = 0;
        for (int i = 0; i < len; i++) {
            if (s.charAt(i) == 'L') {
                lCnt += 1;
            } else {
                rCnt += 1;
            }
            if (lCnt == rCnt) {
                balanceCount += 1;
                lCnt = 0;
                rCnt = 0;
            }
        }
        return balanceCount;
    }
}
// @lc code=end

File: python-vsc/problems/1221.分割平衡字符串.py
#
# @lc app=leetcode.cn id=1221 lang=python3
#
# [1221] 分割平衡字符串
#

# @lc code=start
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        s_len = len(s)
        balance_cnt = 0
        l_cnt, r_cnt = 0, 0
        for i in range(s_len):
            if s[i] == 'L':
                l_cnt += 1
            else:
                r_cnt += 1
            if l_cnt == r_cnt:
                balance_cnt += 1
                l_cnt = 0
                r_cnt = 0
        return balance_cnt

# @lc code=end

